app/src/main/java/org/p2p/wallet/bridge/interactor/EthereumInteractor.kt

```diff
@@ -30,11 +38,21 @@ class EthereumInteractor(
         return ethereumRepository.getPriceForToken(tokenAddress)
     }
 
-    suspend fun loadWalletTokens(ethereumBundleStatuses: List<EthereumClaimToken>) {
-        ethereumRepository.loadWalletTokens(ethereumBundleStatuses)
+    suspend fun loadWalletTokens() {
+        if (ethAddressEnabledFeatureToggle.isFeatureEnabled) {
+            val bundles = getListOfEthereumBundleStatuses()
+            ethereumSendRepository.getSendTransactionsDetail(SolAddress(tokenKeyProvider.publicKey))
+            ethereumRepository.loadWalletTokens(bundles)
+        }
     }
 
-    fun getTokensFlow(): Flow<List<Token.Eth>> = ethereumRepository.getWalletTokensFlow()
+    fun getTokensFlow(): Flow<List<Token.Eth>> {
+        return if (ethAddressEnabledFeatureToggle.isFeatureEnabled) {
+            ethereumRepository.getWalletTokensFlow()
+        } else {
+            flowOf(emptyList())
```

Comment:
is it the expected behavior that a subscriber will receive empty list? You could use `emptyFlow()` instead